src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioHostContextService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    [ExportWorkspaceService(typeof(IHostContextService), ServiceLayer.Host), Shared]
+    internal class VisualStudioHostContextService : IHostContextService
+    {
+        public bool SolutionExistsAndFullyLoaded
+            => KnownUIContexts.SolutionExistsAndFullyLoadedContext.IsActive;
```

Comment:
I'm really concerned that we now have multiple ways of doing this sort of thing.  For example, our delaying of component loading is based on UIContexts.  I feel like we should have just one service that provides the right data for all these systems (ideally written by someone who actually knows how all this solution loading stuff works).  Then we can tie everything to that service instead of having adhoc approaches all of the place that don't actually work for all the differnet solution/project cases out there.